packages/core/src/RenderPipeline/StencilManager.ts

```
@@ -0,0 +1,114 @@
+import { Material } from "../material";
+import { ColorWriteMask, CompareFunction, RenderQueueType, StencilOperation } from "../shader";
+import { DisorderedArray } from "../utils/DisorderedArray";
+import { RenderContext } from "./RenderContext";
+import { RenderQueue } from "./RenderQueue";
+import { SubRenderElement } from "./SubRenderElement";
+
+export class StencilManager {
+  private static _stencilRenderQueue: RenderQueue;
+
+  static getStencilRenderQueue(): RenderQueue {
+    return (StencilManager._stencilRenderQueue ||= new RenderQueue(RenderQueueType.Transparent));
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)